packages/trpc/server/routers/viewer/bookings/get.handler.ts

```diff
@@ -317,13 +318,69 @@ async function getBookings({
               },
             },
           },
+          {
+            eventType: {
+              parent: {
+                team: {
+                  members: {
+                    some: {
+                      userId: user.id,
+                      role: {
+                        in: ["ADMIN", "OWNER"],
+                      },
+                    },
+                  },
+                },
+              },
+            },
+          },
         ],
         AND: [passedBookingsStatusFilter, ...filtersCombined],
       },
       orderBy,
       take: take + 1,
       skip,
     }),
+    // only get personal bookings of current user's team members
```

Comment:
This means that if teampro admin creates managed event type "ABC" for teampro2, and teampro2 creates a private event type "XYZ", and both event types are booked, then teampro admin sees both "ABC" (managed) and "XYZ" (private event created by teampro2)

Here are event types of teampro2 - highlighted is privately created:
<img width="457" alt="Screenshot 2024-02-16 at 10 01 35" src="https://github.com/calcom/cal.com/assets/42170848/bd5da936-7bff-4110-afc3-af7a314ac862">


it was then booked and teampro can see it:
<img width="1510" alt="Screenshot 2024-02-16 at 10 03 42" src="https://github.com/calcom/cal.com/assets/42170848/bc97957e-bdc5-46e0-b21c-a23a53aaa037">

I think that this query should only take personal bookings of user's team members if the event type is managed.

Reply:
the admin should be able to see private bookings of its team members, isn't it?

Reply:
I thought the problem was "Managed" team events. If teampro2 has following event types, then admin should only see managed (green ones) and not see privately created ones (red one) being booked
<img width="457" alt="Screenshot 2024-02-16 at 10 01 35" src="https://github.com/calcom/cal.com/assets/42170848/0a9ef09b-0453-4514-b0c6-f6b615d9b17a">

@PeerRich can you please confirm? I think that if member creates a private event, then he / she would not expect for his / her admin to see them because they are private, not managed.

Reply:
depends. in orgs managers should see all event-types. orgs are not recommended to use for private events

for teams.. hmm. i guess not

Reply:
@SomayChauhan I think maybe it's possible to exclude non-org teams by modifying "// only get personal bookings of current user's team members" query:

<img width="876" alt="Screenshot 2024-02-19 at 09 18 08" src="https://github.com/calcom/cal.com/assets/42170848/a15a4ec2-cec5-44b3-add4-97a4e7b2d1d9">

given that orgs are teams with "parentId: null". What do you think? Need to test this though.

Reply:
@SomayChauhan do you think your code can be modified to implement what Peer wrote above?

Reply:
ofc, taking a look!
can you confirm if i understood things right

# in an org
 - org owner/admin can see all the team-bookings of all the subteams and all the personal bookings of all the users in that org. 
 - team admins/owners can see all team-bookings of their respective subteam and all the personal bookings of their subteam's members.
# in a team
 - team admin/owner cannot see personal bookings of its members but can see the team bookings.

Reply:
yes